src/main/java/org/junit/contrib/java/lang/system/internal/NoExitSecurityManager.java

```diff
@@ -3,37 +3,45 @@
 import java.io.FileDescriptor;
 import java.net.InetAddress;
 import java.security.Permission;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
 
 /**
  * A {@code NoExitSecurityManager} throws a {@link CheckExitCalled} exception
  * whenever {@link #checkExit(int)} is called. All other method calls are
  * delegated to the original security manager.
  */
 public class NoExitSecurityManager extends SecurityManager {
+
 	private final SecurityManager originalSecurityManager;
+	private final long timeout;
+
+	private final CountDownLatch synchLatch = new CountDownLatch(1);
 	private Integer statusOfFirstExitCall = null;
 
-	public NoExitSecurityManager(SecurityManager originalSecurityManager) {
+	public NoExitSecurityManager(SecurityManager originalSecurityManager, long timeout) {
 		this.originalSecurityManager = originalSecurityManager;
+		this.timeout = timeout;
 	}
 
 	@Override
 	public void checkExit(int status) {
 		if (statusOfFirstExitCall == null)
 			statusOfFirstExitCall = status;
+		synchLatch.countDown();
 		throw new CheckExitCalled(status);
 	}
 
-	public boolean isCheckExitCalled() {
-		return statusOfFirstExitCall != null;
+	public boolean isCheckExitCalled() throws InterruptedException {
+		boolean wasCountDown = synchLatch.await(timeout, TimeUnit.MILLISECONDS);
+		return statusOfFirstExitCall != null && wasCountDown;
```

Comment:
@stefanbirkner agreed. Also it's possible to change type of `statusOfFirstExitCall` from `Integer` to simply `int` and remove `if` statement at all here:
```
	@Override
	public void checkExit(int status) {
		if (statusOfFirstExitCall == null)
			statusOfFirstExitCall = status;
		exitLatch.countDown();
		throw new CheckExitCalled(status);
	}
```
WDYT?